File: 8-geometria/1875.hs
import Data.Char (ord)
import Data.Array 

scores::Int->Int->Int
scores issuer receiver
 |issuer==ord('R')&&receiver==ord('G')=2
 |issuer==ord('R')=1
 |issuer==ord('G')&&receiver==ord('B')=2
 |issuer==ord('G')=1
 |issuer==ord('B')&&receiver==ord('R')=2
 |otherwise=1

question::String->Array Int Int->Array Int Int 
question str arr=do
	let inp=words str
	let	first_chr=ord((inp!!0)!!0)
	let	second_chr=ord((inp!!1)!!0)
	let	goals=scores first_chr second_chr

	if first_chr==ord('R')then arr//[(0,((arr!0)+goals))]
	else if first_chr==ord('G')then arr//[(1,(arr!1)+goals)]
	else arr//[(2,(arr!2)+goals)]

answer::Array Int Int->IO(String)
answer goals=do
	if goals!0>goals!1&&goals!0>goals!2 then return "red"
	else if goals!1>goals!0&&goals!1>goals!2 then return "green"
	else if goals!2>goals!0&&goals!2>goals!1 then return "blue"
	else if goals!0==goals!1&&goals!1==goals!2 then return "trempate"
	else return  "empate"

main::IO()
main=do
	n<-readLn::IO Int
	loop_i n

loop_i::Int->IO()
loop_i 0=return ()
loop_i i=do
	m<-readLn::IO Int
	let temp=array(0,2)[(0,0),(1,0),(2,0)]
	goals<-loop_j m temp
	awns<-answer goals
	putStrLn awns
	loop_i (i-1)

loop_j::Int->Array Int Int->IO(Array Int Int)
loop_j 0 arr = return arr
loop_j j arr=do
	temp<-getLine
	let new_arr=question temp arr
	loop_j (j-1) new_arr

File: 8-geometria/1875.c
#include<stdio.h>
#include<string.h>
//
int points(int issuer,int receiver){
	if(issuer==(int)'B'&&receiver==(int)'R')
		return 2;
	else if(issuer==(int)'B')
		return 1;

	if(issuer==(int)'G'&&receiver==(int)'B')
		return 2;
	else if(issuer==(int)'G')
		return 1;

	if(issuer==(int)'R'&&receiver==(int)'G')
		return 2;
	else
		return 1;
}

int main(){
	int rep;
	for(scanf("%i",&rep);rep;--rep){
		int goals,score[]={0,0,0};
		for(scanf("%i",&goals);goals;--goals){
			char team_1[5],team_2[5];
			scanf("%s %s",team_1,team_2);

			if(!strcmp(team_1,"B"))
				score[0]+=points((int)team_1[0],(int)team_2[0]);
			if(!strcmp(team_1,"G"))
				score[1]+=points((int)team_1[0],(int)team_2[0]);
			if(!strcmp(team_1,"R"))
				score[2]+=points((int)team_1[0],(int)team_2[0]);
		}
		if(score[0]>score[1]&&score[0]>score[2])
			printf("blue\n");
		else if(score[1]>score[0]&&score[1]>score[2])
			printf("green\n");
		else if(score[2]>score[0]&&score[2]>score[1])
			printf("red\n");
		else if(score[0]==score[1]&&score[1]==score[2])
			printf("trempate\n");
		else if(score[0]==score[1]||score[0]==score[2]||score[1]==score[2])
			printf("empate\n");
	}

	return 0;
}
